README.md

```diff
@@ -44,8 +44,70 @@ The keypairs are RSA with 2048 bit private keys.
+  * In secure boot mode, the firmware measures into PCR-7 the signer of an EFI application.
+
+**Mikey** owns a key that is in the system's UEFI DB.
+  * owns keys: uefi-db
+  * uefi-db here is a single key that is present in the uefi-db.
+  * EFI applications signed by uefi-db will be executed by the UEFI in SecureBoot mode.
+  * Mikey signs Steven's shim with Mikey's uefi-db key.
+
+**Steven** owns the shim.efi application:
+  * The shim's primary purpose is to validate the kernel that it will execute, ensuring that it is signed by one of the keys in its DB.
+  * Steven's builds shim.efi with these keys in its DB: uki-limited, uki-production, uki-tpm.
+  * The shim measures the signer of the efi application (kernel) into PCR-7
+
+**Arthur** is the Manifest Certificate Authority.
+ * owns keys: manifest-ca
+ * As the Manifest Certificate Authority (manifest-ca), Arthur signs Oliver's Certificate Signing Request (manifest).
+
+**Katherine** owns the kernel.
+  * owns keys: uki-limited, uki-production, uki-tpm.
+  * it is assumed that Katherine has sole access to those 3 keys.
+  * Katherine builds and signs "Universal Kernel Initrd" (UKI) EFI applications.
+  * Katherine signs a UKI with one of uki-limited, uki-production or uki-tpm key.
+  * Katherine includes inside her signed UKI:
+    * a copy of Arthur's public manifest-ca into /manifest-ca.pem.
+    * A 'mos' binary
+
+Oliver owns the operating system
+  * owns keys: manifest, sudi-ca
+  * Oliver CSR with a specific UUID was signed by Arthur.  He owns that UUID prefix.
+  * Oliver generates and signs SUDI certificates for systems.  Those certificates are provisioned to a single machine during provisioning.
+  * Oliver uses his manifest key to sign Manifests (a root filesystem and list of containers).
+
+**Thomas** owns the tpm.
+  * owns keys: tpmpol-luks tpmpol-admin
+  * During provisioning (factory install):
+    * the TPM admin (master password) is set and configured to allow access to via policies that are signed with tpm-admin.
+    * a luks key (randomly generated long key) is generated and stored in the tpm.  Access is granted to policies signed by the tpmpol-luks key.
+    * a SUDI key is registered with the TPM.
+  * Thomas signs policies giving access to
+    * TPM admin password if the booted kernel is signed by Katherine's uki-tpm key.
+    * LUKs key if the booted kernel is signed uki-production.
+
+## Programs
+**Mos binary**.
+  * runs in the initramfs (and after).
+  * during initramfs can access SUDI cert and luks passphrase.
+  * before leaving the initramfs, measures a known constant value into PCR7 so that policies can match against that, and so that policies intended to give access only during initramfs will no longer have access.
+  * Will only pivot into a RFS if:
+    * The manifest is signed by a certificate that has been signed /manifest-ca.pem (from Arthur)
+    * The UUID prefix listed in the system's sudi certificate is owned by the manifest signer.
```

Comment:
(I'm not sure)  ...  And if the serial number in the SUDI cert matches the DMI info?

Reply:
I guess it makes sense to do that, but it does not improve security (I do not think, right?)

Is this True or False: The SUDI key contains both a "product UUID" (Oliver) and a system serial number?

"Olivers UUID" needs a name... i keep wanting to call it product UUID , but i dont really like that name. Reasons for that are;
  * We're just using uuid as a namespace, that shouldn't be in its name.  its really a namespace, so "product namespace" would be better from that perspective.
  * Currently *Oliver* (Operating system) owns that.  I really want to get to a place where there are two roles there, one that owns the OS (rootfs layer and mosctl) and one that owns the non-rootfs layers in the manifest (Connor).  My ultimate goal there would be that no code supplied by Connor ever ran as UID 0.

Reply:
> I guess it makes sense to do that, but it does not improve security (I do not think, right?)

In the VM case where you can just change the qemu arguments, no.  But in a hardware/IOT case, it lets you ensure that only the precise instances you blessed are able to run.

> Is this True or False: The SUDI key contains both a "product UUID" (Oliver) and a system serial number?

True.  See https://github.com/project-machine/trust/blob/main/cmd/trust/sudi.go#L105

> "Olivers UUID" needs a name... i keep wanting to call it product UUID , but i dont really like that name. Reasons for that are;
> 
>     * We're just using uuid as a namespace, that shouldn't be in its name.  its really a namespace, so "product namespace" would be better from that perspective.

Well in our legacy parlance it's "PID", the product ID.

To keep the freedoms we have right now, we must ensure that

a) the 'product UUID" can be provided by the user, and doesn't have to be a UUID.
b) That multiple "product UUID"s can be provided per project.  We have a set of 6 or so PIDs per project, right now.

>     * Currently _Oliver_ (Operating system) owns that.  I really want to get to a place where there are two roles there, one that owns the OS (rootfs layer and mosctl) and one that owns the non-rootfs layers in the manifest (Connor).  My ultimate goal there would be that no code supplied by Connor ever ran as UID 0.

That is indeed (using keyhole as needed) the goal.

Reply:
> In the VM case where you can just change the qemu arguments, no. But in a hardware/IOT case, it lets you ensure that only the precise instances you blessed are able to run.

2 comments here:
1. Hardware DMI isn't necessarily read-only.  It may not be as easy to change serial number in DMI as in qemu, and maybe some hardware has truly read-only serial numbers, but for at least a non-trivial set its just moderately painful (example [thinkpad](https://support.lenovo.com/us/en/solutions/ht501039-how-to-update-the-machine-type-and-model-mtm-system-serial-number-sn-or-system-brand-id-of-system-bios-menu-thinkcentre-thinkstation))
2. It seems to me that to take advantage of this, you have to both have access to a SUDI-cert (which you were not supposed to ever be able to get) *and* then be able to change the serial number.  But if you managed access to SUDI cert, then you can probably load that into a qemu/swtpm setup and just use that to attack further. 

I very well could just be missing something with 2.  I don't think the suggestion is a bad idea, I just don't think it really provides a lot. But I absolutely may be missing something.

Reply:
Some vendors ship products only on hardware where it *is* difficult to change.

We don't want to not support those just because other vendors may ship on hardware where it's not diffucult to change.

Reply:
> We don't want to not support those just because other vendors may ship on hardware where it's not diffucult to change.

ok, so on those platforms what is the *benefit* of doing this?

Reply:
The benefit is that if you are in a known-pristine initrd (as you do), and you see a SUDI which says it is on a secure machine-type, then you know you can proceed.  In contrast, if you see a SUDI which says it is on a secure machine-type, but you see that the machine type does not match, you know something is up.  Extracting the SUDI should not have been possible.

So by extension, if you have a cluster of secure machine-types, and you only allow admission by other machines which have SUDIs saying they are on secure machine-types, you can trust that they *are* in fact on the secure machine-type.  (If they were running on an emulator, faking machine-type, then they would have to use a SUDI extracted from another insecure machine-type instance).